tests/test_phase_psf.py

```diff
@@ -1081,6 +1091,121 @@ def test_withGSP():
     assert optPSF.stepk != optPSF2.stepk
     assert optPSF.drawImage().bounds != optPSF2.drawImage().bounds
 
+
+# Need a non-local picklable function for next test...
+class DummyWork(object):
+    @staticmethod
+    def work(i, atm):
+        rng = galsim.UniformDeviate(i+10)
+        theta = (rng()*galsim.degrees, rng()*galsim.degrees)
+        psf = atm.makePSF(exptime=1.0, diam=1.1, lam=1000.0, theta=theta)
+        return psf.drawImage(nx=32, ny=32, scale=0.2, n_photons=1000, method='phot', rng=rng)
+
+
+@unittest.skipIf(sys.version_info < (3,4), 'Requires python version >= 3.4')
+@timer
+def test_shared_memory():
+    """Test that shared memory hooks to AtmosphericScreen work.
+    """
+    import multiprocessing as mp
+    # Make the atmosphere
+    seed = 12345
+    r0_500 = 0.15  # m
+    L0 = 20.0  # m
+    nlayers = 6
+    screen_size = 102.4  # m
+    screen_scale = 0.1 # m
+    max_speed = 20  # m/s
+    rng = galsim.BaseDeviate(seed)
+    u = galsim.UniformDeviate(rng)
+    Ellerbroek_alts = [0.0, 2.58, 5.16, 7.73, 12.89, 15.46]  # km
+    Ellerbroek_weights = [0.652, 0.172, 0.055, 0.025, 0.074, 0.022]
+    Ellerbroek_interp = galsim.LookupTable(
+            Ellerbroek_alts,
+            Ellerbroek_weights,
+            interpolant='linear')
+    alts = np.max(Ellerbroek_alts)*np.arange(nlayers)/(nlayers-1)
+    weights = Ellerbroek_interp(alts)
+    weights /= sum(weights)
+    spd = []  # Wind speed in m/s
+    dirn = [] # Wind direction in radians
+    r0_500s = [] # Fried parameter in m at a wavelength of 500 nm.
+    for i in range(nlayers):
+        spd.append(u()*max_speed)
+        dirn.append(u()*360*galsim.degrees)
+        r0_500s.append(r0_500*weights[i]**(-3./5))
+    rng2 = rng.duplicate()
+
+    if __name__ == "__main__":
+        ctxs = [None, mp.get_context("fork"), mp.get_context("spawn"), mp.get_context("forkserver")]
+    else:
+        ctxs = [None]
+
+    for ctx in ctxs:
+        if ctx is None:
+            ctx = mp.get_context(None)
+        atm = galsim.Atmosphere(r0_500=r0_500, L0=L0, speed=spd, direction=dirn, altitude=alts,
+                                rng=rng.duplicate(),
+                                screen_size=screen_size, screen_scale=screen_scale,
+                                mp_context=ctx)
```

Comment:
Looks like you're only checking `mp_context` being a `multiprocessing.context.BaseContext`.  I think better to let `None` stay None, since Atmosphere will run `mp.get_context(None)` for you, and let at least one of the others bet the string version.  Say, leave "forkserver" as a string rather than calling `mp.get_context("forkserver")`.